docs/linking/5b-credential-registration.puml

```diff
@@ -0,0 +1,135 @@
+@startuml
+
+!pragma teoz true
+
+title PISP Linking: Credential registration (verification)
+
+participant "PISP" as PISP
+
+box "Mojaloop"
+    participant Switch
+    participant "Auth Service" as Auth
+end box
+
+participant "DFSP" as DFSP
+
+autonumber 10 "<b>CRED-#</b>"
+
+...
+
+note over PISP, DFSP
+    The PISP uses the FIDO registration flow to generate a new keypair and sign the challenge, relying on the user performing an "unlock action" on their mobile device.
+end note
+
+...
+
+PISP -> Switch ++: ""PUT /consents/123 {""\n\
+    ""  requestId: "456",""\n\
+    ""  initiatorId: "PISP",""\n\
+    ""  participantId: "DFSP",""\n\
+    ""  scopes: {""\n\
+    ""    accounts.getBalance: "12345-67890",""\n\
+    ""    accounts.transfer: "12345-67890",""\n\
+    ""  }""\n\
+    ""  credential: {""\n\
+    ""    **id: "9876", // This is new!**""\n\
+    ""    credentialType: "FIDO",""\n\
+    ""    status: "PENDING",""\n\
+    ""    challenge: {""\n\
+    ""      payload: base64(...),""\n\
+    ""      **signature: base64(...), // This is new!**""\n\
+    ""    },""\n\
+    ""    **payload: base64(...), // This is new!**""\n\
+    ""  }""\n\
+    ""}""
+Switch --> PISP: ""202 Accepted""
+deactivate PISP
+
+Switch -> Auth ++: ""PUT /consents/123 {""\n\
+    ""  requestId: "456",""\n\
+    ""  initiatorId: "PISP",""\n\
+    ""  participantId: "DFSP",""\n\
+    ""  scopes: {""\n\
+    ""    accounts.getBalance: "12345-67890",""\n\
+    ""    accounts.transfer: "12345-67890",""\n\
+    ""  }""\n\
+    ""  credential: {""\n\
+    ""    **id: "9876", // This is new!**""\n\
+    ""    credentialType: "FIDO",""\n\
+    ""    status: "PENDING",""\n\
+    ""    challenge: {""\n\
+    ""      payload: base64(...),""\n\
+    ""      **signature: base64(...), // This is new!**""\n\
+    ""    },""\n\
+    ""    **payload: base64(...), // This is new!**""\n\
+    ""  }""\n\
+    ""}""
+Auth --> Switch: ""202 Accepted""
+deactivate Switch
+
+Auth -> Auth: Verify the signature checks out.\nSave the credential.
+Auth -> Switch ++: ""PUT /consents/123 {""\n\
+    ""  requestId: "456",""\n\
+    ""  initiatorId: "PISP",""\n\
+    ""  participantId: "DFSP",""\n\
+    ""  scopes: {""\n\
+    ""    accounts.getBalance: "12345-67890",""\n\
+    ""    accounts.transfer: "12345-67890",""\n\
+    ""  }""\n\
+    ""  credential: {""\n\
+    ""    id: "9876",""\n\
+    ""    credentialType: "FIDO",""\n\
+    ""    **status: "VERIFIED", // This is new!**""\n\
+    ""    challenge: {""\n\
+    ""      payload: base64(...),""\n\
+    ""      signature: base64(...),""\n\
+    ""    },""\n\
+    ""    payload: base64(...),""\n\
+    ""  }""\n\
+    ""}""
+Switch --> Auth: ""200 OK""
+deactivate Auth
+
+Switch -> PISP ++: ""PUT /consents/123 {""\n\
```

Comment:
This is the same as above - we may need the auth service to issue 2 separate `PUT /consents/123`. One to the DFSP and another to the PISP

Reply:
I think, per our discussion the other day, that the "authority" here is the switch rather than the auth service (announcing that consent has been established between two participants on the Mojaloop network). In this case, we'd expect the switch to do the broadcast of the update to the participants involved.

Reply:
Yep. Agreed.